pyface/tasks/action/task_action_manager_builder.py

```diff
@@ -8,50 +8,38 @@
 #
 # Thanks for using Enthought open source!
 
-from collections import defaultdict
 import logging
 
+from traits.api import Instance, List, Property
 
-from pyface.action.api import ActionController, ActionManager
-from traits.api import HasTraits, Instance
-
-
+from pyface.action.schema.action_manager_builder import ActionManagerBuilder
+from pyface.action.schema.schema import ToolBarSchema
+from pyface.action.schema.schema_addition import SchemaAddition
 from pyface.tasks.task import Task
-from pyface.tasks.topological_sort import before_after_sort
-from pyface.tasks.action.schema import Schema, ToolBarSchema
-from pyface.tasks.action.schema_addition import SchemaAddition
+
 
 # Logging.
 logger = logging.getLogger(__name__)
 
 
-class TaskActionManagerBuilder(HasTraits):
-    """ Builds menu bars and tool bars from menu bar and tool bar schema, along
-    with any additions provided by the task.
-    """
+class TaskActionManagerBuilder(ActionManagerBuilder):
+    """ ActionManagerBuilder for Tasks.
 
-    #: The controller to assign to the menubar and toolbars.
-    controller = Instance(ActionController)
+    This provides some additional convenience methods for extracting schema
+    information from a task and using it to build menu bar and toolbar
+    managers directly.
+    """
 
     #: The Task to build menubars and toolbars for.
     task = Instance(Task)
 
+    #: The schema additions provided by the Task.
+    additions = Property(List(SchemaAddition), depends_on='task.extra_actions')
+
     # ------------------------------------------------------------------------
     # 'TaskActionManagerBuilder' interface.
     # ------------------------------------------------------------------------
 
-    def create_action_manager(self, schema):
-        """ Create a manager for the given schema using the task's additions.
-        """
-        additions_map = defaultdict(list)
-        for addition in self.task.extra_actions:
-            if addition.path:
-                additions_map[addition.path].append(addition)
-
-        manager = self._create_action_manager_recurse(schema, additions_map)
-        manager.controller = self.controller
-        return manager
-
     def create_menu_bar_manager(self):
         """ Create a menu bar manager from the task's menu bar schema and
             additions.
```

Comment:
nit: in `create_tool_bar_managers` we go through `self.task.extra_actions` instead of `self.additions`.  They are equivalent, but using `self.additions` feels more consistent

Reply:
It should be changed because if a subclass or other user of the class messes with how the `additions` are built then we will get inconsistent results.